src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -59,7 +66,63 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
         {
             lock (_gate)
             {
-                _openDocumentContexts.Add(e.Document.Id, Task.Run(() => GetConventionContextAsync(e.Document.FilePath, CancellationToken.None)));
+                var documentId = e.Document.Id;
+                var filePath = e.Document.FilePath;
+                _openDocumentContexts.Add(documentId, Task.Run(async () =>
+                {
+                    var context = await GetConventionContextAsync(filePath, CancellationToken.None).ConfigureAwait(false);
+                    context.CodingConventionsChangedAsync += (innerSender, innerE) => HandleCodingConventionsChangedAsync(documentId, innerSender, innerE);
+                    return context;
+                }));
+            }
+        }
+
+        private static readonly ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>> s_projectNotifications =
+            new ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>>();
+
+        private Task HandleCodingConventionsChangedAsync(DocumentId documentId, object sender, CodingConventionsChangedEventArgs e)
+        {
+            var projectId = documentId.ProjectId;
+            var projectsAlreadyNotified = s_projectNotifications.GetOrCreateValue(e);
+            lock (projectsAlreadyNotified)
+            {
+                if (!projectsAlreadyNotified.Add(projectId))
+                {
+                    return Task.CompletedTask;
+                }
+            }
+
+#if EDITORFEATURESWPF
+            var foregroundNotificationService = ((IMefHostExportProvider)_workspace.Services.HostServices).GetExports<IForegroundNotificationService>().Single().Value;
+            foregroundNotificationService.RegisterNotification(UpdateProject, EmptyAsyncToken.Instance, CancellationToken.None);
+#else
+#endif
+
+            return Task.CompletedTask;
+
+            void UpdateProject()
+            {
+                var compilationOptions = _workspace.CurrentSolution.GetProject(projectId)?.CompilationOptions;
+                if (compilationOptions == null)
+                {
+                    return;
+                }
+
+                var parseOptions = _workspace.CurrentSolution.GetProject(projectId)?.ParseOptions;
+                if (parseOptions.Features.TryGetValue("EditorConfigWorkaround", out _))
+                {
+                    parseOptions = parseOptions.WithFeatures(parseOptions.Features.Where(feature => feature.Key != "EditorConfigWorkaround"));
+                }
+                else
+                {
+                    parseOptions = parseOptions.WithFeatures(parseOptions.Features.Concat(new[] { KeyValuePair.Create("EditorConfigWorkaround", "true") }));
+                }
+
+                _workspace.OnCompilationOptionsChanged(projectId, compilationOptions);
```

Comment:
I dont think we ever update these not coming from project system.

Reply:
➡️ Project system has no support for this feature, and we determined during our last design meeting that it will not be possible to complete project system support for this feature during the 15.8 timeline. We already agreed to investigate alternative approaches as an interim solution. This would be one of those approaches.

Considering it is inevitable for one or more items in the interim solution to be suboptimal, we should consider how best to document those areas for future cleanup.